lambdas/backend-api/src/templates/app/get-template.ts

```diff
@@ -0,0 +1,38 @@
+import { ITemplateClient } from 'nhs-notify-backend-client';
+import { templateRepository } from '@backend-api/templates/domain/template';
+import { userRepository } from '@backend-api/templates/domain/user';
+import { logger, success } from '@backend-api/utils/index';
+
+export const getTemplate: ITemplateClient['getTemplate'] = async (
+  templateId,
+  token
+) => {
+  const userResult = await userRepository.getUser(token);
```

Comment:
Instead of doing this, you can get the authorizer to do this (the authorizer does do this already in fact), and add it to the request context (it doesn't do this bit). Let me dig something out on the subject.

Reply:
That would be great! I thought we could but didn't look into it TBH. 

IIRC, you can return context variables from the authorizer lambda.

Reply:
https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-lambda-authorizer-output.html

Basically the authorizer lambda already checks the JWT and even calls Cognito to check that the user exists.

In the return value of the authorizer, you can return something like this:

```ts
return {
  principalId: 'api-caller',
  policyDocument: {
    Version: '2012-10-17',
    Statement: [
      {
        Action: 'execute-api:Invoke',
        Effect,
        Resource,
      },
    ],
  },
  context: {
    user: JSON.stringify({ id: user.id })
  }
}
```

That `context` object gets passed downstream by API Gateway into the handler lambda. And you can use it like `event.requestContext.authorizer.user`

It's not a massive thing I don't think. Because you aren't actually making a network call to Cognito or anything to get the user, just parsing it from the JWT. But having it in the authorizer means you can have the same logic once, and all your auth/jwt logic in one place where it belongs. Rather than doing it in each Lambda, and the actual endpoint lambdas don't have to concern themselves with JWTs and other auth-related things.

Reply:
I guess you'd still want to Zod your `event.requestContext.authorizer.user` - but you're doing that anyway

Reply:
Updated. to just pass the username in the authorizer context.